nautobot_floor_plan/static/nautobot_floor_plan/js/label_utils.js

```diff
@@ -0,0 +1,145 @@
+function generateLabels(start, end, step, labelType, incrementLetter) {
+    let labels = [];
+    
+    if (labelType === "numbers") {
+        // Numeric labels
+        let startNum = parseInt(start, 10);
+        let endNum = parseInt(end, 10);
+        for (let i = startNum; i <= endNum; i += step) {
+            labels.push(i.toString().padStart(2, '0')); // Pad with leading zeros if necessary
+        }
+    } else if (labelType === "letters") {
+        // Letter labels
+        let currentCharCode = start.charCodeAt(0);
+        let endCharCode = end.charCodeAt(0);
+        while (currentCharCode <= endCharCode) {
+            labels.push(String.fromCharCode(currentCharCode));
+            currentCharCode += step;
+        }
+    } else if (labelType === "alphanumeric" || labelType === "numalpha") {
+        // Alphanumeric or Numalpha labels
+        let numericPart = start.match(/\d+/)[0]; // Extract the numeric part
+        let startLetters = start.match(/[A-Za-z]+/)[0]; // Extract the letter part
+        let endLetters = end.match(/[A-Za-z]+/)[0]; // Extract the end letter part
+
+        let currentLetters = startLetters;
+        while (true) {
+            labels.push(`${numericPart}${currentLetters}`);
+            if (incrementLetter) {
+                currentLetters = incrementString(currentLetters, step);
+                if (currentLetters > endLetters) break;
+            } else {
+                currentLetters = incrementWholeString(currentLetters, step);
+                if (currentLetters > endLetters) break;
+            }
+        }
+    } else if (labelType === "roman") {
+        // Roman numeral labels
+        let startNum = romanToInt(start);
+        let endNum = romanToInt(end);
+        for (let i = startNum; i <= endNum; i += step) {
+            labels.push(intToRoman(i));
+        }
+    } else if (labelType === "greek") {
+        // Greek letter labels
+        let startNum = greekToInt(start);
+        let endNum = greekToInt(end);
+        for (let i = startNum; i <= endNum; i += step) {
+            labels.push(intToGreek(i));
+        }
+    } else if (labelType === "binary") {
+        // Binary labels
+        let startNum = parseInt(start, 10);
+        let endNum = parseInt(end, 10);
+        for (let i = startNum; i <= endNum; i += step) {
+            labels.push(intToBinary(i));
+        }
+    } else if (labelType === "hex") {
+        // Hexadecimal labels
+        let startNum = parseInt(start, 16);
+        let endNum = parseInt(end, 16);
+        for (let i = startNum; i <= endNum; i += step) {
+            labels.push(intToHex(i));
+        }
+    }
```

Comment:
Might be good to have an `else` case that reports an error, just thinking of what might happen if a new case is added server-side but we forget to add it here?